src/classes/RD_ProcessControl.cls

```diff
@@ -38,11 +38,28 @@ public with sharing class RD_ProcessControl {
     /*******************************************************************************************************
     * @description flag to prevent recursion into the RD trigger handler/logic
     */ 
-    public static boolean hasRun = false;
+    private static Integer recursionFlag = 0;
```

Comment:
It doesn't appear that static methods are supported in Virtual/Abstract base classes, more specifically they are not inherited by the derived class. Basically, I can define a static method in the Virtual or Abstract base class, but that method is not visible in a class that Extends that base class.